### PR TITLE
Add support for AnsiBSTR.

### PR DESCRIPTION
**Pull request by kant2002**
Related to #167

### CODE REVIEW

**Comment by kant2002** [comment]
This is just a test that failure happens with current implementation. Will provide implementation soon. I just want to better understand that this is proper way how to mark items as implemented in the build

**Comment by kant2002** [comment]
@jkotas or @MichalStrehovsky anybody knows why this is green? In my naivete, I expect this is to fail.

**Comment by jkotas** [comment]
Per-PR CI runs only the smoke tests.

File: src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/InteropHelpers.cs [hunk 1]
```diff
@@ -231,6 +231,31 @@ internal static unsafe string BstrBufferToString(char* buffer)
...
+            if (s is null)
+            {
+                return (byte*)IntPtr.Zero;
+            }
+
+            int stringLength = s.Length;
+            fixed (char* pStr = s)
+            {
+                int nativeLength = PInvokeMarshal.GetByteCount(pStr, stringLength);
+                byte* bstr = (byte*)Marshal.AllocBSTR(nativeLength);
```
Comment (jkotas) [inline src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/InteropHelpers.cs]: I do not think that this works. `AllocBSTR` takes length in characters, but you are passing it length in bytes.

File: src/tests/issues.targets [hunk 2]
```diff
@@ -1172,9 +1172,6 @@
         <ExcludeList Include="$(XunitTestBinBase)/Interop/PInvoke/Variant/VariantTest/*">
             <Issue>https://github.com/dotnet/runtimelab/issues/155: VARIANT marshalling</Issue>
         </ExcludeList>
-        <ExcludeList Include="$(XunitTestBinBase)/Interop/StringMarshalling/AnsiBSTR/AnsiBStrTest/*">
```
Comment (jkotas) [inline src/tests/issues.targets:1175]: Did you run this test locally and verified that it is passing?

**Comment by MichalStrehovsky** [comment]
/azp run runtimelab-nativeaot Pri-0 Tests

This should trigger the P0 test legs. I don't expect them to all pass, but we can see how the test in question is doing.

**Comment by kant2002** [comment]
My apologies. Do not have time today to run the build. Will check how it is going

**Comment by kant2002** [comment]
Even if Pri0 tests fails in unrelated areas, when I run locally, it fails. So I will fix that.

File: src/tests/issues.targets (hunk 2 above)
Comment (kant2002) [reply src/tests/issues.targets:1175]: Yes. I do verify now.

**Review by jkotas — APPROVED** [review]
Thanks

**Pull request merged**